### PR TITLE
Speed improvements part 2: general.js

### PR DESCRIPTION
**Pull request by baltpeter**
This PR seriously cuts down on the stuff that we load on every page.

Ready for review, @zner0L.

### CODE REVIEW

**Comment by baltpeter** [comment]
This PR just expanded in scope a little to also improve our error handling a bit. I would have opened a separate PR instead, but this PR modifies the error handler quite a bit, so that wouldn't have made sense…

**Review by zner0L — CHANGES_REQUESTED** [review]
Thanks a lot!

File: src/error-handler.js [hunk 1]
```diff
@@ -9,11 +9,40 @@ function logError(event) {
...
+        </a>
+        <a id="error-modal-email-link" class="button button-small">
+            ${I18N_DEFINITION['error-handler']['report-via-email']}
+        </a>
+    </div>
+</div>`;
+
+    // Using innerHTML is a little naughty but it is fine here, since we don't include anything that could possibly be
+    // attacker-controlled. All that content is later added safely.
+    document.body.innerHTML += html;
```
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/error-handler.js:42]: As far as I remember, innerHTML is actually [faster](https://www.quirksmode.org/dom/innerhtml.html), at least compared to the DOM methods, not sure about Preact.

Either way, I agree this isn't an issue here. Showing an error modal should definitely be an exception that only occurs once in a blue moon and that potentially taking a couple ms longer seems like a reasonable trade-off to me compared to bloating every page load.

File: src/Components/DonationWidget.js [hunk 2]
```diff
@@ -372,3 +372,9 @@ export default class DonationWidget extends preact.Component {
         );
     }
 }
+
+window.renderDonationWidget = function () {
```
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/Components/DonationWidget.js:376]: That's a pattern I have adopted a while ago. While not necessary in this case, it allows us to export other functions in this file and then import them somewhere else without the render being triggered.

I agree that this is not… clean. From what I saw, we might be able to get rid of that once the JS is built by Hugo using [resources from string](https://gohugo.io/hugo-pipes/resource-from-string/)?

File: src/error-handler.js [hunk 3]
```diff
@@ -9,11 +9,40 @@ function logError(event) {
     /* eslint-enable no-console */
 }
 
-try {
-    const preact = require('preact');
-    const Modal = require('Components/Modal').default;
-    const t = require('Utility/i18n').default;
```
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/error-handler.js]: I don't think. We are losing fields and plurals but other than that, I believe `t()` does the exact same thing that I inlined here. It also works directly on the `I18N_DEFINITION`.

Falling back to the English strings is implemented in the Webpack loader.

**Review by zner0L — DISMISSED** [review]

**Review by zner0L — APPROVED** [review]

**Pull request merged**